icubam/messaging/sender.py

```diff
@@ -0,0 +1,57 @@
+from absl import logging
+
+from icubam.messaging import sms_sender
+from icubam.messaging.telegram import integrator
+from icubam.messaging import email_sender
+
+
+class Sender:
+  """The Sender class is responsible for sending messages to the users.
+
+  It decides the channel to be used to contact the user: sms, email or
+  via telegram bot depending on the user itself and the configuration of
+  the server.
+  """
+  def __init__(self, config, db, queue, tg_bot=None):
+    self.config = config
+    self.db = db
+    self.queue = queue
+
+    try:
+      self.sms_sender = sms_sender.get(config)
+    except Exception as e:
+      logging.warning(f"Cannot set sms_sender {e}")
+      self.sms_sender = None
+
+    try:
+      self.email_sender = email_sender.get(config)
+    except Exception as e:
+      logging.warning(f"Cannot set email_sender {e}")
+      self.email_sender = None
+
+    # This might return be None if telegram is not properly set.
+    self.telegram_bot = integrator.TelegramSetup(config, db, tg_bot=tg_bot).bot
+
+  async def process(self):
+    """Keeps reading the sending queue and does send the messages."""
+    async for msg in self.queue:
+      try:
```

Comment:
This function is never called in tests.

Reply:
yeah... i am not sure yet how to unit test it, but i ll give it a try.